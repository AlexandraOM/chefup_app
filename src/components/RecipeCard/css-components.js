import React from "react";
import styled from "styled-components";

const ImageContainer = styled.div`
  background-image: url(${props => props.imageUrl});
  background-size: cover;
  background-position: center;
  height: 150px;
  width: 100%;
`;

export { ImageContainer };
