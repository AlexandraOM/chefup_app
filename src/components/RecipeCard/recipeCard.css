section {
  max-width: 100vh;
  display: flex;
  justify-content: space-between;
  flex-flow: row wrap;
}

article {
  max-width: 80vh;
  display: flex;
  padding: var(--space-m);
  margin: var(--space-m);
}
.recipes_lesson-details {
  border-radius: 10px;
  border-color: var(--grey);
  border-width: 10px;
  max-width: 80vh;
  background-color: var(--white);
  padding: var(--space-m);
  margin: var(--space-m);
}

.recipe_image {
  height: auto;
  max-height: 250px;
  width: auto;
  max-width: 250px;
}
