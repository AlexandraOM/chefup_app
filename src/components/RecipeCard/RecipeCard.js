import React from "react";
import "./recipeCard.css";
import { getRecipesData } from "../../utils/getRecipesData";
import styled from "styled-components";

const ImageContainer = styled.div`
  background-image: url(${props => props.imageUrl});
  background-size: cover;
  background-position: center;
  height: 150px;
  width: 100%;
`;

const RecipeCard = () => {
  const [lessonData, setLessonData] = React.useState(null);

  React.useEffect(() => {
    getRecipesData().then(data =>
      setLessonData(
        data.records.map(lesson => ({
          title: lesson.fields.lesson_name,
          description: lesson.fields.lesson_description,
          teacher: lesson.fields.teacher,
          time: lesson.fields.time,
          duration: lesson.fields.lesson_duration,
          image_url: lesson.fields.main_recipe_image[0].url,
          ingredients: lesson.fields.ingredients,
          id: lesson.id
        }))
      )
    );
  }, []);

  if (lessonData === null) {
    return (
      <div>
        <p>Loading...</p>
      </div>
    );
  }

  // lessonData.forEach(element => {
  //   // console.log('lesson description ', element.fields.lesson_description)
  //   console.log('lesson description', element.lesson_description);
  // });
  //{" <div className="recipe_image-container">"}
  //{"</div> "}
  return (
    <section className="recipes">
      {lessonData
        ? lessonData.map(lesson => (
            <article data-testid={lesson.id} className="recipes_recipeCard">
              <div className="recipes_lesson-details">
                <ImageContainer
                  imageUrl={lesson.image_url}
                  alt={lesson.title}
                ></ImageContainer>

                <h1>{lesson.title}</h1>
                <h2>{lesson.teacher}</h2>
                <p>{lesson.description}</p>
                <p>{lesson.time}</p>
                <p>{lesson.duration}</p>
                <p>{lesson.ingredients}</p>
              </div>
            </article>
          ))
        : null}
    </section>
  );
};

export default RecipeCard;
