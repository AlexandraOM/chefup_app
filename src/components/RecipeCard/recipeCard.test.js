import React from "react";
import { render } from "@testing-library/react";
import RecipeCard from "./RecipeCard";

test("RecipeCard component renders correctly", () => {
  const { findByTestId } = render(<RecipeCard />);
  findByTestId("rec8oKeNrianbq65v");
});
