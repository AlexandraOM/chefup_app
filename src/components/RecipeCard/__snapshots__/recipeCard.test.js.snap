// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer matches the snapshot 1`] = `
<div>
  <p>
    Details: enter here
  </p>
  <p>
    My recipe details
  </p>
</div>
`;
